sdk/trace/src/main/scala/org/typelevel/otel4s/sdk/trace/SpanProcessor.scala

```diff
@@ -143,19 +189,49 @@ object SpanProcessor {
     private val endOnly: List[SpanProcessor[F]] =
       processors.filter(_.isEndRequired)
 
+    val name: String =
+      s"SpanProcessor.Multi(${processors.map(_.name).mkString_(", ")})"
+
     def isStartRequired: Boolean = startOnly.nonEmpty
     def isEndRequired: Boolean = endOnly.nonEmpty
 
     def onStart(parentCtx: Option[SpanContext], span: SpanRef[F]): F[Unit] =
-      startOnly.parTraverse_(_.onStart(parentCtx, span))
+      startOnly
+        .parTraverse { p =>
+          p.onStart(parentCtx, span).attempt.tupleLeft(p.name)
+        }
+        .flatMap(attempts => handleAttempts(attempts))
```

Comment:
this is a bit nitpicky (also below), but

```suggestion
        .flatMap(handleAttempts)
```